### PR TITLE
Feat/#21_#57 search for plants - Testing

### PR DESCRIPTION
**Pull request by tobireith**
- Fix Example Test, add config for running all tests
- Adjust Search to testing. Implement basic test for search functionality
- Remove unused stuff and TODO

### CODE REVIEW

**Review by emuguy1 — APPROVED** [review]
In general, you shouldn't import multiple View Models into One Activity. The way to do this would be, to add the database connection into your own View Model but other than that. Looks great to me.

File: app/src/main/java/de/othr/plantico/ui/search/SearchViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package de.othr.plantico.ui.search
+
+import androidx.lifecycle.ViewModel
+import de.othr.plantico.database.entities.Plant
+
```
Comment (hoffmatteo) [inline app/src/main/java/de/othr/plantico/ui/search/SearchViewModel.kt]: This ViewModel does not make sense, a ViewModel should hold the data itself. If you look at PlantViewModel, the ViewModel is created by a ViewModelFactory which passes the database repository, and the ViewModel holds the plant lists. If the data is passed by the PlantViewModel anyway, I don't see the reason for another ViewModel.

**Review by hoffmatteo — CHANGES_REQUESTED** [review]

File: app/src/main/java/de/othr/plantico/ui/search/SearchViewModel.kt (hunk 1 above)
Comment (tobireith) [reply app/src/main/java/de/othr/plantico/ui/search/SearchViewModel.kt]: Deleted the ViewModel and put the search function to the PlantViewModel as discussed.
Please have a look again :)

File: app/src/main/java/de/othr/plantico/ui/PlantViewModel.kt [hunk 2]
```diff
@@ -20,6 +20,13 @@ class TestViewModel(private val repository: Repository) : ViewModel() {
     fun insertOwnedPlant(plant: OwnedPlant)= viewModelScope.launch {
         repository.insertOwnedPlant(plant)
     }
+
+    companion object {
+        //Search for plants that contain the query as a substring. Not case-sensitive!
+        fun searchForPlantsInList(plants: List<Plant>, query: String): List<Plant> {
+            return plants.filter { plant: Plant -> plant.plantName.contains(query, ignoreCase = true) }
+        }
+    }
```
Comment (hoffmatteo) [inline app/src/main/java/de/othr/plantico/ui/PlantViewModel.kt]: This does not make sense. A ViewModel contains the data, in this case the plantList, and passes it to the Activity. Here, you reverse this order, which is unnecessary as the ViewModel already has access to the plantList. Either use the ViewModel's data here, or move it to the Activity. In the Activity it shouldn't be static as well, as you should use the plantList property you defined there.

**Review by hoffmatteo — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 3]
```diff
@@ -52,6 +52,7 @@ dependencies {
     implementation 'androidx.lifecycle:lifecycle-livedata-ktx:2.4.1'
     implementation 'androidx.lifecycle:lifecycle-viewmodel-ktx:2.4.1'
     testImplementation 'junit:junit:4.13.2'
+    testRuntimeOnly 'org.junit.vintage:junit-vintage-engine:5.8.2'
```
Comment (emuguy1) [inline app/build.gradle]: warum hast du da jUnit5 nochmal drinnen

**Review by emuguy1 — COMMENTED** [review]
Agree to the comment of Matteo

**Review by hoffmatteo — APPROVED** [review]

**Pull request merged**